### PR TITLE
docs for new MINID trim strategy

### PR DESCRIPTION
**Pull request by guybe7**

### CODE REVIEW

File: commands/xtrim.md [hunk 1]
```diff
@@ -10,10 +13,20 @@ the latest 1000 items:
...
 
+To give another exmaple, the following command will trim the stream
+so that it won't contain entries with IDs smaller than 649085820:
+
+```
+XTRIM mystream MINID 649085820-0
+```
+
 By default, or when provided with the optional `=` argument, the command
-performs exact trimming. That means that the trimmed stream's length will be
```
Comment (oranagra) [inline commands/xtrim.md:14]: here's a good example of why starting a new sentence in an existing line causes an unnecessary diff

**Review by oranagra — COMMENTED** [review]
sorry for hogging this PR to discuss the new style guides and evaluate their pros and cons.

File: commands.json [hunk 2]
```diff
@@ -4682,7 +4682,8 @@
         "type": "block",
         "block": [
           {
-            "command": "MAXLEN"
+            "name": "trim-strategy",
+            "type": "string"
```
[the remark this answers is not in the document]
Comment (itamarhaber) [reply commands.json:4686]: Yep, this is my edit:
image

File: commands.json [hunk 3]
```diff
@@ -4697,10 +4698,16 @@
                 "optional": true
               },
               {
-                "name": "length",
-                "type": "integer"
+                "name": "threshold",
+                "type": "string"
```
[the remark this answers is not in the document]
Comment (itamarhaber) [reply commands.json:4702]: Regrettably, our syntax has implied parenthesis and no way to explicitly state that. Yet another point for future improvement.

[the remark this answers is not in the document]
Comment (oranagra) [reply commands.json:4702]: @itamarhaber why can't we do this:
```
XTRIM key [MAXLEN [=|~] length] | [MINID [=|~] id]
```
and then state somewhere that you must provide at least, and only one.

i'm not certain it's better than
```
XTRIM key (MAXLEN|MINID) [=|~] threshold
```
just suggesting

**Comment by itamarhaber** [comment]
Syntax preview:

image
image

File: commands/xtrim.md [hunk 4]
```diff
@@ -22,19 +35,31 @@ order to make it more efficient:
...
+the user is not really requesting that the stream is trimmed exactly at the `threshold`,
+but instead it could be a few tens of entries more, but never less than the `threshold`.
+When this option modifier is used, the trimming is performed only when
 Redis is able to remove a whole macro node. This makes it much more efficient,
 and it is usually what you want.
 
+The `LIMIT` option represents the maximum number of entries to trim, and so offers
+another way to cap the trimming, making it more efficient on the expense of accuracy.
+This option is only valid with the `~` modifier and its default is (100 * the number
+of entries per macro node).
```
[the remark this answers is not in the document]
Comment (oranagra) [reply commands/xtrim.md]: @itamarhaber please respond to this one.

**Review by oranagra — APPROVED** [review]

File: commands/xadd.md [hunk 5]
```diff
@@ -48,26 +48,15 @@ IDs to match the one of this other system.
...
-and the specified maximum length.
-
-Trimming with **MAXLEN** can be expensive compared to just adding entries with 
-`XADD`: streams are represented by macro nodes into a radix tree, in order to
-be very memory efficient. Altering the single macro node, consisting of a few
-tens of elements, is not optimal. So it is possible to give the command in the
-following special form:
+`XADD` incorporates the same semantics as the `XTRIM` command - refer to its documentation page for more information.
+This allows adding new entries and keeping the stream's size in check with a single call to `XADD`, effectively capping the stream with an arbitrary threshold.
+Although exact trimming is possible and is the default, due to the internal representation of steams it is more effecient to add an entry and trim stream with `XADD` using **almost exact** trimming (the `~` argument).
```
Comment (oranagra) [inline commands/xadd.md:53]:
```suggestion
Although exact trimming is possible and is the default, due to the internal representation of steams it is more efficient to add an entry and trim stream with `XADD` using **almost exact** trimming (the `~` argument).
```

**Review by oranagra — APPROVED** [review]
